### PR TITLE
Refactoring APIPurge and IDPPurge into a new DAO

### PR DESCRIPTION
**Pull request by rashm1n**
$Subject

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeSQLConstants.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.wso2.carbon.apimgt.cleanup.service;
```
Comment (HiranyaKavishani) [inline components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeSQLConstants.java:1]: License header missing

File: components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+     * Method to get the instance of the ApiMgtDAO.
+     *
+     * @return {@link OrganizationPurgeDAO} instance
+     */
+    public static OrganizationPurgeDAO getInstance() {
+
+        if (INSTANCE == null) {
+            INSTANCE = new OrganizationPurgeDAO();
+        }
+
```
Comment (HiranyaKavishani) [inline components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java:63]: Remove unnecessary line

File: components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+     * @throws APIManagementException
+     */
+    public void deleteOrganizationAPIList(String organization) throws APIManagementException {
+
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            connection.setAutoCommit(false);
+
+            // Remove records from AM_API table and associated data through cascade delete
+            String deleteAPIQuery = OrganizationPurgeSQLConstants.REMOVE_BULK_APIS_DATA_FROM_AM_API_SQL;
+            deleteOrganizationAPIData(connection, deleteAPIQuery, organization);
```
Comment (HiranyaKavishani) [inline components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java:113]: use deleteAPIQuery inside deleteOrganizationAPIData

Comment (HiranyaKavishani) [reply components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java:113]: Follow same in other

File: components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+            }
+        } catch (SQLException e) {
+            log.error("Error while getting apiUuid list of organization" + orgId, e);
+            handleException("Failed to get API apiUuid list of organization " + orgId, e);
+        }
+        return apiList;
+    }
+
+    /**
+     * Delete all organization API data
```
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java:101]: make sure to add a `.` at the end of first sentense in javadoc description. this causes checkstyle failures

File: components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                }
+                preparedStatement.execute();
+                conn.commit();
+            } catch (SQLException e) {
+                conn.rollback();
+                throw e;
+            }
+        } catch (SQLException e) {
+            throw new APIManagementException("Error while deleting key managers:  " + kmIdList + " in organization "
+                    + organization,e);
```
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/OrganizationPurgeDAO.java:204]: shouldn't you use handleException here aswell?

**Pull request merged**